src/components/Column/Column.tsx

```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { css, cx } from "emotion";
+
+const flexRowStyles = css`
+  display: flex;
+  flex-direction: column;
+`;
+
+export const Column = ({
+  children,
+  className
+}: {
+  children: React.ReactNode;
+  className?: string;
```

Comment:
it may be a common pattern in libs, but I do not see it as a work around. I feel that if it does exactly what it says then it is an ok abstraction.

My vision for `Column` and `Row` is to mimic `div` HTML elements, for the moment the only prop we would need is classNames, but we could also support useRef's and other props in the future.

I also do not fully understand what you are referring to as an alternative solution, I am guessing you mean we should just have styles applied in a child wrapper? ex:
Currently:
```
<Row className={someNewClass}>
    ...
</Row>
```

Your suggestion:
```
<Row>
   <div className={someNewClass}>
    ...
    </div>
</Row>
```

Feel free to correct me if i misunderstand.

I would be interested in getting @gtrufitt take on this approach

Reply:
So the issue with passing 'className' as a prop is it essentially lets you do anything. It's basically the `any` of props and as such should be used as a last resort when you have no other choice.

But we have other choices, we control the code. The problem here is you want to make `Row` work for multiple use cases with different styles.
So we could:

1. Have style specific props, like `isFullWidth` or `backgroundColor` etc.
2. Have different components, like `FullWidthRow` or `BlueRow`
3. Use `styled` like `const BlueRow = styled(Row)`background: blue`
4. Keep `Row` basic and have a second div with the context specific styles applied to it
5. Not abstract at all, just define the components needed locally

My preference would be to have a basic `Row` component if there are several places in the code that need it and there's a use case for it. But where the requirements are to have different styles applied then we should manage them them locally, either by extra divs or by just adding the css needed as normal, not using Row because Row isn't what we want in that situation.

We've hit this situation multiple times in the code here and on DCR, where a seemingly generic component turned out to have subtle use cases needing extra styles and in all those situations we've not resorted to `className` (except for Adslot) and I'm super reticent to start adding this pattern now.

Reply:
To be clear, my dislike of the `className` pattern is that it does not scale. The whole point of react and typescript is to keep code maintainable as it scales but when you let people start passing classNames around you lose control and readability and your code rapidly turns into spaghetti

Reply:
Hmm, I guess I'm slightly split in this context - having `className` on layout elements seems like a reasonable thing and I've certainly wished for it before rather than wrapping elements.

However, on balance, I agree with Oliver purely for the reason that it keeps the layout components clear in their responsibility, and if we think that they need extra functionality passing it as props is a nicer UX as auto-completable. 

It also keeps styling that we might otherwise be tempted to pass through, creating a mis-direction, close to the component that's using it.

Reply:
Ok yeah, Oliver replied as I was typing, he's a faster typer.

Reply:
To give some context for my passion on this point. I worked on a project that used this pattern and I grew to Utterly Hate It. I'd be constantly tracing `className` props though the code to try and see where styles were coming from and repeatedly ended up lost and frustrated. My stress levels rose significantly which may or may not have shortened my lifespan. All because of `className` props.

I spent a large part of my time refactoring it away which was slow and painful because you had to relearn all the use cases and break the abstractions.

I _get_ that it's tempting. It's tempting because it's easy, as soon as you add it all your problems go away and life is great. But it's a pact with the devil, you are deferring the pain for later, probably to someone else, maybe me, and I'm don't want to go through that ever again.

Reply:
@oliverlloyd I am coming around to the idea of keeping both components minimalist.

I have another suggestion., maybe we could make the `Row` and `Column` just a flex-box abstraction? By adding common flex-box options the we use `justify-content: space-between;` and `width: 100%`

Or would this go counter to your point?

Reply:
My *final* point: Why is this okay for styles but not for other types of prop? if you think about it, it's the same as just having `randomProps`  If you're going to say `className` you might as well call it `someRandomStylesWhahayLetsGo` and then you can call the other props `randomPropsYeah` and just spread that untyped prop love everywhere.

Reply:
> By adding common flex-box options the we use `justify-content: space-between;` and `width: 100%`

Yep, sure. This makes sense and is readable. I have zero issues with this. 👍 

There is that classic risk of reimplementing the flex api - most people have been there - but I'm pretty sure if you open most mature react codebases you will find similar components and there's nothing wrong with them

Reply:
This has been a great discussion, thanks for sharing @oliverlloyd. 

I would love to have a similar abstraction in dotcom-rendering.

Reply:
Thanks for this, really interesting. I guess DCR has https://github.com/guardian/dotcom-rendering/blob/master/src/web/components/Flex.tsx